R/method-differentialAbundance.R

```diff
@@ -275,6 +282,25 @@ setMethod("differentialAbundance", signature("AbundanceData", "Comparator"), fun
       stop('Unaccepted differential abundance method. Accepted methods are "DESeq" and "Maaslin".')
     }
     veupathUtils::logWithTime(paste0('Completed method=',method,'. Formatting results.'), verbose)
+
+    # Sometimes p-values can be very small, even smaller than the smallest representable number (gives p-value=0). The smallest
+    # representable number changes based on env, so to avoid inconsistency set a p-value floor so that any
+    # returned p-value less than the floor becomes the floor. 
+    # The default floor is a constant defined in the microbiomeComputations package.
+
+    # First set small pvalues to the pValueFloor
+    statistics@statistics[statistics@statistics$pValue < pValueFloor, "pValue"] <- pValueFloor
+
+    # Second, find the adjusted p value for this floored p-value
+    flooredPValues <- c(pValueFloor, statistics@statistics[statistics@statistics$pValue > pValueFloor, "pValue"])
+    adjustedPValuesWithFloor <- stats::p.adjust(flooredPValues)
+    adjustedPValueFloor <- adjustedPValuesWithFloor[1]
+
+    # Finally, update the adjusted p-values in the df
+    statistics@statistics[statistics@statistics$pValue == pValueFloor, "adjustedPValue"] <- adjustedPValueFloor
```

Comment:
am i missing something? this looks like youre setting an adjusted pvalue for the floored values only, rather than all of them?

Reply:
right, i'm setting it for the floored values only. I didn't think it appropriate to recalculate for all of the values since i want all but the floored values to stay true to whatever the original value was. With FDR it shouldn't change but still it seems better to leave those values as they were originally calculated

Reply:
Can we confirm that the other values are unaffected? I understand why they should be identical, but without actually using the same implementation in both places I'd just like us to be sure.